### PR TITLE
MODINV-184 - Implement action handler for Create Item action

### PR DESCRIPTION
**Pull request by RuslanLavrov**
https://issues.folio.org/browse/MODINV-184

### Approach:
* implemented action handler for Create Item action.
*  published event about Item creation using data-import-processing-core

### CODE REVIEW

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    ArrayList<String> errorMessages = new ArrayList<>();
+    for (String fieldPath : requiredFields) {
+      String field = StringUtils.substringBefore(fieldPath, ".");
+      String nestedField = StringUtils.substringAfter(fieldPath, ".");
+      if (!isExistsRequiredProperty(itemAsJson, field, nestedField)) {
+        errorMessages.add(String.format("Field '%s' is a required field and can not be null", fieldPath));
+      }
+    }
+    String statusName = JsonHelper.getNestedProperty(itemAsJson, "status", "name");
+    if (StringUtils.isNotBlank(statusName) && !ItemStatusName.isStatusCorrect(statusName)) {
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java]: Status is also required field, so `status == null` is not allowed.

File: src/test/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+  @Mock
+  private Storage mockedStorage;
+  @Mock
+  private ItemCollection mockedItemCollection;
+  @Mock
+  private Reader fakeReader;
+  @Spy
+  private MarcBibReaderFactory fakeReaderFactory = new MarcBibReaderFactory();
+
+  protected JobProfile jobProfile = new JobProfile()
```
Comment (bohdan-suprun) [inline src/test/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandlerTest.java]:
```suggestion
  private JobProfile jobProfile = new JobProfile()
```

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java (hunk 1 above)
Comment (RuslanLavrov) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java]: if nested field ` name != null ` then ` status != null ` as well, therefore Item with status == null will not be created

File: src/test/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandlerTest.java (hunk 2 above)
Comment (RuslanLavrov) [reply src/test/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandlerTest.java]: thank you

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java]: Yep, sorry, I meant that `status.name == null` is also unacceptable. See https://github.com/folio-org/mod-inventory-storage/blob/master/ramls/item.json#L245.

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  @Override
+  public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload dataImportEventPayload) {
+    CompletableFuture<DataImportEventPayload> future = new CompletableFuture<>();
+    try {
+      HashMap<String, String> payloadContext = dataImportEventPayload.getContext();
+      if (payloadContext == null || StringUtils.isBlank(payloadContext.get(EntityType.MARC_BIBLIOGRAPHIC.value()))) {
+        LOG.error(PAYLOAD_HAS_NO_DATA_MSG);
+        future.completeExceptionally(new EventProcessingException(PAYLOAD_HAS_NO_DATA_MSG));
+        return future;
+      }
```
Comment (VRohach) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java]: @RuslanLavrov please add some tests for this case

Comment (RuslanLavrov) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java]: already added

**Pull request merged**